spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -252,12 +266,12 @@ public WriteBuilder newWriteBuilder(LogicalWriteInfo info) {
     Preconditions.checkArgument(
         snapshotId == null, "Cannot write to table at a specific snapshot: %s", snapshotId);
 
-    return new SparkWriteBuilder(sparkSession(), icebergTable, info);
+    return new SparkWriteBuilder(sparkSession(), icebergTable, branch, info);
   }
 
   @Override
   public RowLevelOperationBuilder newRowLevelOperationBuilder(RowLevelOperationInfo info) {
-    return new SparkRowLevelOperationBuilder(sparkSession(), icebergTable, info);
+    return new SparkRowLevelOperationBuilder(sparkSession(), icebergTable, branch, info);
```

Comment:
https://github.com/apache/iceberg/blob/master/spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java#L290 SparkTable#canDeleteUsingMetadata needs to be updated to use the branch as well during the scan